llm/prompt.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import sys
+
+from langchain.prompts import PromptTemplate
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.prompts import MessagesPlaceholder
+
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+
+basic_question_prompt_template = """You are a helpful assistant and an expert on the internal administration, personnel and projects of the innovation agency Nesta. Use the following pieces of context to extract facts which help answer the question. Keep your replies short and informative. If the context does not help you answer the question, reply 'NULL'.
+    {context}
+    Question: {question}
+    """  # noqa
+
+basic_question_prompt = PromptTemplate(
+    template=basic_question_prompt_template, input_variables=["context", "question"]
+)
+
+
+qa_system_prompt = """You are a helpful assistant and an expert on the internal administration, personnel and projects of the innovation agency Nesta. Use the following pieces of context to extract facts which help answer the question. Keep your replies short and informative. If the context does not help you answer the question, reply 'NULL'.
+    {context}
+    """  # noqa
+
+qa_prompt = ChatPromptTemplate.from_messages(
+    [
+        ("system", qa_system_prompt),
+        MessagesPlaceholder("chat_history"),
+        ("human", "{input}"),
+    ]
+)
+
+
+# credit: https://medium.com/@eric_vaillancourt/mastering-langchain-rag-integrating-chat-history-part-2-4c80eae11b43
+contextualize_q_system_prompt = """Given a chat history and the latest user question \
+which might reference context in the chat history, formulate a standalone question \
+which can be understood without the chat history. Do NOT answer the question, \
+just reformulate it if needed and otherwise return it as is."""
```

Comment:
This prompt is copied from examples I have seen of incorporating chat history into RAG systems, for example: https://python.langchain.com/v0.1/docs/use_cases/question_answering/chat_history/. Its purpose is to rephrase the user input into a query for the retriever. In a chat the user input often makes references to concepts in the chat history by using words like pronouns or "that" or "this" instead of the actual concept – which would be problematic if used directly as query for the retriever. This prompt may not be optimal and we could experiment with others (low priority though I guess).